src/app/core/pending-changes.guard.ts

```diff
@@ -0,0 +1,20 @@
+import { CanDeactivate } from '@angular/router';
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs';
+
+export interface ComponentCanDeactivate {
+  canDeactivate: () => boolean | Observable<boolean>;
+}
+
+
+@Injectable()
+export class PendingChangesGuard implements CanDeactivate<ComponentCanDeactivate> {
```

Comment:
@TommyHuang-dev could you add proper TS Documentation to this canDeactivate function instead of having the comments inside the function?